src/server/server.ts

```diff
@@ -152,33 +176,54 @@ export class CTProtoServer<AuthRequestPayload, AuthData, ApiRequest extends NewM
    * @param data - message data
    */
   private async onmessage(socket: ws, data: ws.Data): Promise<void> {
-    try {
-      MessageValidator.validateMessage(data as string);
-    } catch (error) {
-      const errorMessage = (error as Error).message;
-
-      this.log(`Wrong message accepted: ${errorMessage} `, data);
-
-      if (error instanceof CriticalError) {
-        socket.close(CloseEventCode.UnsupportedData, errorMessage);
-      } else {
-        socket.send(MessageFactory.createError('Message Format Error: ' + errorMessage));
-      }
+    await this.validateTextMessage(data, socket);
```

Comment:
I think, the `validateTextMessage` should not accept the `socket`. It should carry only about the single task: validation. And throw exception in cases of invalid message